602 feature request show professor email

Troy-bailan-King opened the pull request:
Able to show professor name and email on Coursepage and students can add/drop course on Coursepage. If user want to see couse on Schedule page, need to refresh this page.

![image](https://user-images.githubusercontent.com/97757033/229229244-3ca8dc0a-9fb3-412a-92f6-1c3e48273ba8.png)
![image](https://user-images.githubusercontent.com/97757033/229229266-9bcfb94a-e00c-4dad-901f-657cbaf11e7f.png)
![image](https://user-images.githubusercontent.com/97757033/229229290-785d4b4b-5294-4b82-a677-581a576dbfbe.png)

jonha1 reviewed (changes requested):
Great idea adding the name of the professor. The feature would be better if the user shouldn't have to refresh the page to see the schedule.

Troy-bailan-King commented:
Yeah, but I have no idea why the schedule doesn't show in time.

chavab0809 reviewed (changes requested):
Tried to review this pull request however I got this error when I opened the website.
![Screenshot 2023-04-07 155931](https://user-images.githubusercontent.com/123212886/230671010-f7952247-ad5c-4c5a-9c77-fc4bbc80e825.png)

Troy-bailan-King commented:
> Tried to review this pull request however I got this error when I opened the website. ![Screenshot 2023-04-07 155931](https://user-images.githubusercontent.com/123212886/230671010-f7952247-ad5c-4c5a-9c77-fc4bbc80e825.png)

Fix the problem

zhengw5 reviewed (changes requested):
I think it is a pretty useful feature to add because many students email their professors during registration due to conflicts or issues that come up. Choosing the section in this page is also pretty useful but I agree that the schedule should be updated without needing to refresh the webpage.

zhangn7 reviewed (changes requested):
The feature works in that it will show the professor's name next to each section they teach. However, when we put the section on the schedule display, we cannot scroll down to see which professor teaches that section if the class block is too short to fit all the information.

mlammon11 reviewed (dismissed):
This feature is a good addition to the website, however I found a bug where if you search the course, for example CSCI 2500, and add it to you schedule, if you click on the hyperlink to see course details, the professors don't show up.

mlammon11 reviewed (changes requested):
Mostly works but for some professors there isn't the correct first and last name, and there's not email. Ex David Goldschmidt teaches CSCI 2500, so If I click on CSCI 2500, "Goldsc" is displayed with no email

Troy-bailan-King commented:
> Mostly works but for some professors there isn't the correct first and last name, and there's not email. Ex David Goldschmidt teaches CSCI 2500, so If I click on CSCI 2500, "Goldsc" is displayed with no

Troy-bailan-King commented:
> Goldsc



> Mostly works but for some professors there isn't the correct first and last name, and there's not email. Ex David Goldschmidt teaches CSCI 2500, so If I click on CSCI 2500, "Goldsc" is displayed with no email

Because There are many different name abbreviations that exist in different course databases, I have now added this abbreviation to the professor information and fixed the previous hyperlink problem, try again.

Troy-bailan-King commented:
> I think it is a pretty useful feature to add because many students email their professors during registration due to conflicts or issues that come up. Choosing the section in this page is also pretty useful but I agree that the schedule should be updated without needing to refresh the webpage.

Now we don't need to refresh

Troy-bailan-King commented:
> The feature works in that it will show the professor's name next to each section they teach. However, when we put the section on the schedule display, we cannot scroll down to see which professor teaches that section if the class block is too short to fit all the information.

Somebody did it on different PR

mlammon11 commented:
> 

Yes the hyperlink problem is resolved, but what do you mean by "I have now added this abbreviation to the professor information"? I tried running your branch again and I still see "Goldsc"

Troy-bailan-King commented:
Which semester are you testing for

mlammon11 commented:
> Which semester are you testing for

Spring 2022

Troy-bailan-King commented:
> > Which semester are you testing for
> 
> Spring 2022

<img width="1262" alt="image" src="https://user-images.githubusercontent.com/97757033/233736802-8d357543-5cf7-4ad8-82f3-f1318223a111.png">

mlammon11 commented:
> > > Which semester are you testing for
> > 
> > 
> > Spring 2022
> 
> <img alt="image" width="1262" src="https://user-images.githubusercontent.com/97757033/233736802-8d357543-5cf7-4ad8-82f3-f1318223a111.png">

Oop i guess when I refreshed it I forgot to pull from origin, I can see it now. However, there are other courses facing the same issue such as "ARCHITECTURAL DESIGN STUDIO 2" displays "Sehwail" with no email.

Troy-bailan-King commented:
This is because the names of professors in the course database are not stored according to the norm, for example: Goldschmidt is abbreviated as Goldsc and Goldschmidt in different courses. this problem may have to be reorganized in the database, now only one can be found, then one can be added.
